pkg/scheduler/framework/plugins/volumebinding/binder_test.go

```diff
@@ -333,55 +333,61 @@ func (env *testEnv) initVolumes(cachedPVs []*v1.PersistentVolume, apiPVs []*v1.P
 
 }
 
-func (env *testEnv) updateVolumes(t *testing.T, pvs []*v1.PersistentVolume, waitCache bool) {
-	for _, pv := range pvs {
-		if _, err := env.client.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{}); err != nil {
+func (env *testEnv) updateVolumes(t *testing.T, pvs []*v1.PersistentVolume) {
+	for i, pv := range pvs {
+		if newPv, err := env.client.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{}); err != nil {
 			t.Fatalf("failed to update PV %q", pv.Name)
+		} else {
+			pvs[i] = newPv
 		}
 	}
-	if waitCache {
-		wait.Poll(100*time.Millisecond, 3*time.Second, func() (bool, error) {
-			for _, pv := range pvs {
-				obj, err := env.internalPVCache.GetAPIObj(pv.Name)
-				if obj == nil || err != nil {
-					return false, nil
-				}
-				pvInCache, ok := obj.(*v1.PersistentVolume)
-				if !ok {
-					return false, fmt.Errorf("PV %s invalid object", pvInCache.Name)
-				}
-				if versioner.CompareResourceVersion(pvInCache, pv) != 0 {
-					return false, nil
-				}
+	err := wait.Poll(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+		for _, pv := range pvs {
+			obj, err := env.internalPVCache.GetAPIObj(pv.Name)
+			if obj == nil || err != nil {
+				return false, nil
 			}
-			return true, nil
-		})
+			pvInCache, ok := obj.(*v1.PersistentVolume)
+			if !ok {
+				return false, fmt.Errorf("PV %s invalid object", pvInCache.Name)
+			}
+			if versioner.CompareResourceVersion(pvInCache, pv) != 0 {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to update PV %v", err)
```

Comment:
My 2 cents: given the function is named `updateVolumes`, it's a bit aggressive to fatal the test directly. I'd suggest one of the two options:

1. Return the err, and let the callers to decide how to respond with the err (@ahg-g pointed out: https://github.com/kubernetes/kubernetes/pull/109043/files?w=1#r846922035)
2. Compose two functions: `updateXYZ` and `udpateXYZOrDie`. The latter one calls the previous one and fatal the test if an err is returned. On caller side, we can call `udpateXYZOrDie`.

Reply:
Modify the code according to option one.

Reply:
> My 2 cents: given the function is named `updateVolumes`, it's a bit aggressive to fatal the test directly. I'd suggest one of the two options:
> 
> 1. Return the err, and let the callers to decide how to respond with the err (@ahg-g pointed out: https://github.com/kubernetes/kubernetes/pull/109043/files?w=1#r846922035)
> 2. Compose two functions: `updateXYZ` and `udpateXYZOrDie`. The latter one calls the previous one and fatal the test if an err is returned. On caller side, we can call `udpateXYZOrDie`.

Modify the code according to option 1